### PR TITLE
Commvault Cloud: Bugfix : Commvault Cloud for Splunk SOAR

### PR DESCRIPTION
**Pull request by ghost**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by ishans-crest** [comment]
@Cv-securityIQ  Can you please squash old commits that got carried from previous PR and also fix merge conflicts?

**Comment by Commvault-3rdPartyIntegrations** [comment]
Hi @ishans-crest , I've squashed the commits and fixed merge conflicts. Can you please review the PR ?

File: release_notes/unreleased.md [hunk 1]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Initial release
```
Comment (ishans-crest) [inline release_notes/unreleased.md]: Please update the release note mentioning the bug that has been fixed.

**Comment by stutib-crest** [comment]
Hello @Cv-securityIQ ,
In the commvaultcloud_consts.py file are ANOMALY_TYPE_2_PAYLOAD and ANOMALY_TYPE_3_PAYLOAD, authorization tokens? If yes, we suggest to add them to the configuration parameter.

**Comment by Commvault-3rdPartyIntegrations** [comment]
> Hello @Cv-securityIQ , In the commvaultcloud_consts.py file are ANOMALY_TYPE_2_PAYLOAD and ANOMALY_TYPE_3_PAYLOAD, authorization tokens? If yes, we suggest to add them to the configuration parameter.

Hi @ishans-crest. It's not actually a token. It's a JSON that we encoded as a string to make it more readable. We encoded it to JSON because it's a huge JSON string, and we plan to add more types of payloads in the future. This will help keep the constants file readable. Additionally, we didn't want to expose the JSON payload directly.

**Review by hardik-crest — APPROVED** [review]

**Review by ishans-crest — APPROVED** [review]

**Comment by stutib-crest** [comment]
Hi @Cv-securityIQ ,
We noticed that 'fetch_anomalous_events' is added as an action id in the handle_action method in the commvaultcloud_connector.py file, but it is not mentioned in the json file as an action. The _fetch_anomalous_events function is used in the on_poll action, so is it meant to be a separate action?

**Comment by stutib-crest** [comment]
> Hi @Cv-securityIQ , We noticed that 'fetch_anomalous_events' is added as an action id in the handle_action method in the commvaultcloud_connector.py file, but it is not mentioned in the json file as an action. The _fetch_anomalous_events function is used in the on_poll action, so is it meant to be a separate action?

Hello @Cv-securityIQ , 
Could you please look into this?

**Comment by Commvault-3rdPartyIntegrations** [comment]
Hi @stutib-crest , currently we don't have any plans to use it as a separate action.

**Pull request merged**